controllers/utils/common.go

```diff
@@ -32,25 +34,37 @@ func GetMinOf3(number1, number2, number3 int) int {
 }
 
 // GetSafeResourceName returns the safename if already allocated in the map and creates a new one if not
-func GetSafeResourceName(name string, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, maxLength, spareLength int) string {
+func GetSafeResourceName(name, namespace string, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, maxLength int, log *logr.Logger) string {
 	if clusterGroupUpgrade.Status.SafeResourceNames == nil {
 		clusterGroupUpgrade.Status.SafeResourceNames = make(map[string]string)
 	}
-	safeName, ok := clusterGroupUpgrade.Status.SafeResourceNames[name]
+	safeName, ok := clusterGroupUpgrade.Status.SafeResourceNames[namespace+"/"+name]
+
 	if !ok {
-		safeName = NewSafeResourceName(name, clusterGroupUpgrade.GetAnnotations()[NameSuffixAnnotation], maxLength, spareLength)
-		clusterGroupUpgrade.Status.SafeResourceNames[name] = safeName
+		safeName = NewSafeResourceName(name, namespace, clusterGroupUpgrade.GetAnnotations()[NameSuffixAnnotation], maxLength, log)
+		clusterGroupUpgrade.Status.SafeResourceNames[namespace+"/"+name] = safeName
+	} else {
+		if log != nil {
+			log.Info("safename already in  clusterGroupUpgrade.Status.SafeResourceNames",
+				"safename", safeName,
+				"namespace", namespace,
+				"safeName+namespace length <= 62", utf8.RuneCountInString(safeName+namespace))
```

Comment:
Nit, if we have to keep this log, drop the "<= 62" part.

Reply:
fixed